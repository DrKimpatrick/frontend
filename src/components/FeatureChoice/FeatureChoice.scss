@media (max-width: 768px) {
  .feature-choice-section {
    margin-bottom: 50px;
    .title {
      margin-left: 0;
      padding: 0 10px;
      width: 100%;
      text-align: center;
    }
    .back-arrow {
      display: none;
    }
  }
}
