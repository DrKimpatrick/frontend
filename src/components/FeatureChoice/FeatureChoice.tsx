import React from 'react';
import useSWR from 'swr';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import { useHistory } from 'react-router-dom';
import { MainBackground, NavBar, SplashScreen } from 'components/Reusable';
import { Talent } from 'components/Home';

const FeatureChoice = () => {
  const { data: productResponse } = useSWR('/stripe/products');
  const history = useHistory();

  if (!productResponse) {
    return <SplashScreen />;
  }

  return (
    <>
      <NavBar />
      <section className="container mx-auto px-12 flex flex-col items-center justify-center">
        <div className="text-2xl flex items-center text-gray-texts">
          <ArrowBackTwoToneIcon
            className="cursor-pointer"
            onClick={() => history.goBack()}
          />
          <h1 className="ml-4 font-bold">
            What features would you like to have?
          </h1>
        </div>

        <div className="mt-8 w-full">
          {productResponse && (
            <div className="flex flex-col items-center md:justify-between lg:items-start lg:flex-row">
              {productResponse.products.map((product: any) => {
                let productAction = '/notification';
                if (product.name.includes('Standard')) {
                  productAction = '/standard-billing';
                } else if (product.name.includes('Premium')) {
                  productAction = '/premium-billing';
                }
                return (
                  <Talent
                    key={product.id}
                    productDetails={product.productDetails}
                    plans={product.plans}
                    action={productAction}
                  />
                );
              })}
            </div>
          )}
        </div>
      </section>
      <MainBackground />
    </>
  );
};

export default FeatureChoice;
