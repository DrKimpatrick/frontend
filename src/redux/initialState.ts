import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
