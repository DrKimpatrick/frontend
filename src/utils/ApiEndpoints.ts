export const ApiEndPoint = {
  addMcqQuestion: `/question/mcq`,

  deleteQuestion: (questionId: string) => `/question/mcq/${questionId}`,

  UploadVideo: '/upload/video',

  test: '/test',

  verifiedQuestions: '/question/filter/Verified',

  addVideoQuestion: '/question/video'
};
