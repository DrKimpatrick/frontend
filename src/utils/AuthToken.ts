import jwtDecode from 'jwt-decode';

export type DecodedToken = {
  readonly userId: string;
  readonly roles: string[];
  readonly iat: number;
  readonly exp: number;
};

export default class AuthToken {
  readonly decoded: DecodedToken;

  constructor(readonly token?: string) {
    this.decoded = {
      userId: '',
      roles: [''],
      iat: 0,
      exp: 0
    };

    try {
      if (token) this.decoded = jwtDecode(token);
    } catch (e) {
      // TODO: Log to error tracking
    }
  }

  get expiresAt(): Date {
    return new Date(this.decoded.exp * 1000);
  }

  get isExpired(): boolean {
    return new Date() > this.expiresAt;
  }

  get isAuthenticated(): boolean {
    return !this.isExpired;
  }

  get bearerString() {
    return `Bearer ${this.token}`;
  }
}

export const getAccessToken = () => {
  return localStorage.getItem('token');
};
