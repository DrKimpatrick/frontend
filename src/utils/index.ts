import jwt from 'jsonwebtoken';

type errorTypeRegister = {
    username: string;
    email: string;
    password: string;
}

type errorTypeLogin = {
    email: string;
    password: string;
}

type errorAddEmploy = {
    companyName : string;
    title: string;
    responsibilities: string;
    accomplishments: string;
    favoriteProject: string;
    supervisor: string;
    date: string;
    skillsUsed: string
}

export const validEmailRegex = RegExp(
    /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i
);

export const passwordRegex = RegExp(
    /^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@$!%*?&])([a-zA-Z0-9@$!%*?&]{8,})$/
)

export const validateForm = (errors:errorTypeRegister | errorTypeLogin | errorAddEmploy) => {
    let valid = true;
    Object.values(errors).forEach(val => val.length > 0 && (valid = false));
    return valid;
  };

export const decodeToken = (token: any) => {
    const decodedToken = jwt.decode(token)

    return decodedToken
}