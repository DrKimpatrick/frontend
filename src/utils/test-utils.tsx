import React, { FC, ReactElement } from 'react';
import { render } from '@testing-library/react';
import { SWRConfig } from 'swr';

const Wrapper: FC = ({ children }) => {
  return <SWRConfig value={{ dedupingInterval: 0 }}>{children}</SWRConfig>;
};

const customRender = (ui: ReactElement, options: any) =>
  render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
