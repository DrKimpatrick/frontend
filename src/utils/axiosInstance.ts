import axios from 'axios';
import createAuthRefreshInterceptor from 'axios-auth-refresh';
import AuthToken, { getAccessToken } from './AuthToken';

const axiosInstance = axios.create({
  baseURL: process.env.REACT_APP_BACKEND_API,
  headers: {
    'Content-Type': 'application/json'
  }
});

axiosInstance.interceptors.request.use(async config => {
  const token = getAccessToken();

  // eslint-disable-next-line dot-notation
  if (token) config.headers['Authorization'] = `Bearer ${token}`;

  return config;
});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const refreshAuthLogic = async (failedRequest: any) => {
  const refreshToken = localStorage.getItem('ttlnt.refresh') || null;

  return axios
    .get('/auth/refresh', {
      baseURL: process.env.REACT_APP_BACKEND_API,
      headers: {
        Authorization: `Bearer ${refreshToken}`
      }
    })
    .then(resp => {
      const { token: newToken } = resp.data;
      const decoded = new AuthToken(newToken);

      // eslint-disable-next-line dot-notation
      failedRequest.response.config.headers['Authorization'] =
        decoded.bearerString;

      localStorage.setItem('token', newToken);
    })
    .catch(() => {
      // User is logged out when refresh token expires
      localStorage.removeItem('token');
      localStorage.removeItem('ttlnt.refresh');
      // window.location.href = Routes.Login;
    });
};

createAuthRefreshInterceptor(axiosInstance, refreshAuthLogic, {
  pauseInstanceWhileRefreshing: true
});

const fetcher = async (url: string) => {
  const res = await axiosInstance.get(url);
  return res.data;
};

export { fetcher };
export default axiosInstance;
