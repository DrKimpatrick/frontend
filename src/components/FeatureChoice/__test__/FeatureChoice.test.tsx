import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, cleanup } from 'utils/test-utils';
import { cache } from 'swr';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import FeatureChoice from '../FeatureChoice';
import initialState from 'redux/initialState';

describe(FeatureChoice, () => {
  let store: any;
  const middleware = [thunk];
  const mockStore = configureMockStore(middleware);

  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(() => {
    cache.clear();
    cleanup();
  });

  it('matched the snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <FeatureChoice />
        </Router>
      </Provider>,
      {}
    );
    expect(tree).toMatchSnapshot();
  });
});
