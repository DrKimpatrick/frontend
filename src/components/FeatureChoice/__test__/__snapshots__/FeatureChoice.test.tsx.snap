// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureChoice matched the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex items-center bg-white-800 px-20 py-5 flex-wrap"
      >
        <a
          class="text-gray-800"
          href="/"
        >
          <img
            alt="logo"
            class="logo w-24"
            src="logo-image.png"
          />
        </a>
        <div
          class="hidden top-navbar w-full lg:inline-flex lg:flex-grow lg:w-auto"
          id="navigation"
        >
          <div
            class="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
          >
            <div>
              <a
                class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
                href="/"
              >
                <span>
                  How it works
                </span>
              </a>
              <a
                class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
                href="/"
              >
                <span>
                  About us
                </span>
              </a>
            </div>
            <a
              class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
              href="/login"
            >
              <span>
                Login
              </span>
            </a>
            <button
              class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded items-center justify-center bg-blue-700 hover:bg-blue-800 text-white"
              type="button"
            >
              <span>
                Get started
              </span>
            </button>
          </div>
        </div>
      </nav>
      <section
        class="container mx-auto px-12 flex flex-col items-center justify-center"
      >
        <div
          class="text-2xl flex items-center text-gray-texts"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root cursor-pointer"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          <h1
            class="ml-4 font-bold"
          >
            What features would you like to have?
          </h1>
        </div>
        <div
          class="mt-8 w-full"
        />
      </section>
      <div
        class="mainBackground"
      >
        <img
          alt=""
          src="motif-image.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex items-center bg-white-800 px-20 py-5 flex-wrap"
    >
      <a
        class="text-gray-800"
        href="/"
      >
        <img
          alt="logo"
          class="logo w-24"
          src="logo-image.png"
        />
      </a>
      <div
        class="hidden top-navbar w-full lg:inline-flex lg:flex-grow lg:w-auto"
        id="navigation"
      >
        <div
          class="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
        >
          <div>
            <a
              class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
              href="/"
            >
              <span>
                How it works
              </span>
            </a>
            <a
              class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
              href="/"
            >
              <span>
                About us
              </span>
            </a>
          </div>
          <a
            class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/login"
          >
            <span>
              Login
            </span>
          </a>
          <button
            class="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded items-center justify-center bg-blue-700 hover:bg-blue-800 text-white"
            type="button"
          >
            <span>
              Get started
            </span>
          </button>
        </div>
      </div>
    </nav>
    <section
      class="container mx-auto px-12 flex flex-col items-center justify-center"
    >
      <div
        class="text-2xl flex items-center text-gray-texts"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root cursor-pointer"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
        <h1
          class="ml-4 font-bold"
        >
          What features would you like to have?
        </h1>
      </div>
      <div
        class="mt-8 w-full"
      />
    </section>
    <div
      class="mainBackground"
    >
      <img
        alt=""
        src="motif-image.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
